### ISSUE TITLE
[Bug]: Live streams do not appear on subscriptions page nor on channel page

### ISSUE THREAD

**Issue by bpiec** [FreeTubeApp/FreeTube#2852]
### Guidelines

- [X] I have encountered this bug in the [latest release of FreeTube](https://github.com/FreeTubeApp/FreeTube/releases).
- [X] I have searched the issue tracker for [open](https://github.com/FreeTubeApp/FreeTube/issues?q=is%3Aopen+is%3Aissue) and [closed](https://github.com/FreeTubeApp/FreeTube/issues?q=is%3Aissue+is%3Aclosed) issues that are similar to the bug report I want to file, without success.
- [X] I have searched the [documentation](https://docs.freetubeapp.io/) for information that matches the description of the bug I want to file, without success.
- [X] This issue contains only one bug.

### Describe the bug

There was recently a live stream on one of the channels. Youtube shows it like that:

![obraz](https://user-images.githubusercontent.com/1611112/201068234-eb49cce8-caa4-4777-89f5-deeae1ff720c.png)
(it says 13 hours ago).

FreeTube does not show it at. This is a channel page:
![obraz](https://user-images.githubusercontent.com/1611112/201068642-55e38c79-7f90-4b1f-acdc-5662524ce4c6.png)
This is sorted by newest. The first movie is a scheduled one, the second one is from 6 days ago. There is no the live stream from 13 hours ago. I am able to search for it however, when I put its name in the search box on this channel page.

This video is also not shown in the subscription page in FreeTube.

The possible cause of this is that YT now has a separate section for live streams:

![obraz](https://user-images.githubusercontent.com/1611112/201069199-d7a55f6a-6c2c-4ff0-8e73-f0f0915364eb.png)

I don't really care about the LIVE section added as long the live streams appear in the Subscriptions view…

### Expected Behavior

The video should be visible in the Subscriptions page and probably also in the Channel view.

### Issue Labels

API issue

### FreeTube Version

v0.18.0 Beta

### Operating System Version

Windows 11 22H2 22621.819

### Installation Method

.exe

### Primary API used

Local API

### Nightly Build

- [X] I have encountered this bug in the latest [nightly build](https://docs.freetubeapp.io/development/nightly-builds).

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
Also happening to shorts. Previously we were fetching everything from the videos tab on the channel page and all kind of videos were in there (Normal videos, lives streams/VOD and shorts). All those videos were also being fetched like that to be displayed in your subscriptions.

Now we are doing the same as before, fetching from the videos tab and if ur subscribed to that channel it will also display like that in your subscriptions. Shorts and live streams/VODs will not be displayed now.

Whats changed is that YT separated the shorts and VODs/live streams to its own section/tab. We are currently not fetching the videos located in the Live and Shorts section. Therefore u will not see them in your subscriptions page.

**Comment by bpiec** [comment]
I get it. But are there plans to change that?
I also saw the Shorts being missing but I don’t care about them 😛

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
Yes there are plans to address that but it will happen when there is time for it. We are all contributors that are spending their free time on this.

So yeah TLDR: yes, it happens when we have time for it.

**Comment by absidue** [comment]
The big issue with implementing something like that, is that you would be trippling the number of requests to YouTube, so you are a lot more likely to get ratelimited. FreeTube currently fetches the subscriptions from the RSS feeds once you have 125 or more subscriptions, as the RSS feeds don't have any rate limiting (on YouTube's side, Invidious might be different). If we were implement it and tripple the number of requests (one each for the videos, livestreams and shorts tabs vs one for the videos tab) we would probably need to massively reduce that limit to something as low as 42.

We can consider implementing it but we would need to do it in a way that is still user friendly. One idea I've just thought of is that we could implement separate Videos, Live and Shorts tabs on the subscriptions page, that way you would be less likely to get ratelimited, because you would only be using one tab at a time.
No ETA though.

**Comment by My-Random-Thoughts** [comment]
Further to @absidue suggestion, maybe have an option to set specific channels to download one of the three options.
I have one specific channel that only have live shows, so there would be no point looking for normal videos or shorts.

**Comment by mkalinski** [comment]
FWIW, streams and shorts do still appear after switching "Fetch Feeds from RSS" on, so it's a possible workaround until this issue is resolved.

**Comment by arrowgent** [comment]
youtube changed these
but importantly does invidious currently have a fix for this?
currently invidious does not, it only shows Uploaded videos.  no live etc.


videos are split into:
Uploads, Shorts, Live

feature removed:
Sort by: newest, oldest, popular.  replaced by Recently Uploaded & Popular


freetube depends on both youtube and invidious to parse information

**Comment by testertv** [comment]
Version v0.13.2 Beta shows ALL video, including short and streaming. But it is very old and does not work very well at all.

Also I noticed that this issue happens if you put stream in a playlist before/during starting the stream. Then it doesn't visible in the "Videos" section of YouTube and consequently, in the program.

P.S.: The same issue is observed on NewPipe and has not been fixed!

**Comment by elshimone** [comment]
> The big issue with implementing something like that, is that you would be trippling the number of requests to YouTube, so you are a lot more likely to get ratelimited. FreeTube currently fetches the subscriptions from the RSS feeds once you have 125 or more subscriptions, as the RSS feeds don't have any rate limiting (on YouTube's side, Invidious might be different). If we were implement it and tripple the number of requests (one each for the videos, livestreams and shorts tabs vs one for the videos tab) we would probably need to massively reduce that limit to something as low as 42.
> 
> We can consider implementing it but we would need to do it in a way that is still user friendly. One idea I've just thought of, is that we could implement separate Videos, Live and Shorts tabs on the subscriptions page, that way you would be less likely to get ratelimited, because you would only be using one tab at a time. No ETA though.

@absidue I could have a go at this if you like?  Your solution of separating out the different video types and lazy loading them when each tab is accessed sounds like it should not be too tricky?

**Comment by elshimone** [comment]
I've been looking at how we would retrieve the information for shorts and live streams.  Looking at the yt-channel-info component, this has been recently updated to include a support for fetching shorts and live streams (https://github.com/FreeTubeApp/yt-channel-info/commit/496900f995f663b73e9d16a5a322a68d597fe4e3). Unfortunately these changes have not been released yet.

A further wrinkle is that the RSS feed does include shorts at least, but there is no metadata to indicate "this is a short" in the feed entry.  This makes it hard to provide a consistent UI in both cases, as we can't for example filter out the shorts from the videos tab.

**Comment by elshimone** [comment]
Looks like invidious has added support for shorts recently https://github.com/iv-org/invidious/pull/3374 but this is also not yet released.  I think it makes sense to work on this when support for fetching shorts and livestreams is released in all API options.

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
yeah we really should do a release of the modules soon

**Comment by ChunkyProgrammer** [comment]
@elshimone See my comment here: https://github.com/FreeTubeApp/FreeTube/issues/2476#issuecomment-1379451659 😄

**Comment by BarbzYHOOL** [comment]
We need this!!!
btw i hope we can hide these stupid shorts

**Comment by jungle-boogie** [comment]
Many of the latter comments here are discussing the ability to show and filter shorts, so I thought I would make an update to reflect what was originally asked for and what I found out - show live videos in channel page.

The comment by [mkalinski](https://github.com/mkalinski) does allow live videos to appear in your subscriptions, but for all your channels.   The per channel live feature is missing from `FreeTube`.

Invidious does appear to have an option to show live videos for the channel:
https://invidious.privacydev.net/channel/UCY1kMZp36IQSyNx_9h4mpCg/streams

As it looks like shorts per channel filtering is being worked on in #2476, hopefully this will be an easy implementation as well.

**Comment by gonzoknows** [comment]
neither do youtube shorts

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
@elshimone i think u can continue to work on this.

> I've been looking at how we would retrieve the information for shorts and live streams. Looking at the yt-channel-info component, this has been recently updated to include a support for fetching shorts and live streams (FreeTubeApp/yt-channel-info@496900f). Unfortunately these changes have not been released yet.

Local API has been fully migrated to [YouTube.js](https://github.com/LuanRT/YouTube.js/) so u dont have to wait on a release of that module.

**Comment by hcgrove** [comment]
The workaround suggested by @mkalinski does add a lot of videos to your "Subscriptions", but it has the negative consequence that the duration of the video is not shown on the "Subscriptions" page, so I can't see how long a video is before starting it.

**Comment by hcgrove** [comment]
The workaround suggested by @mkalinski does add a lot more videos to my "Subscriptions" page, but (as documented) the RSS feeds for channels don't include e.g. video's duration, and I use that quite a lot, so I would prefer a dedicated fix to this issue.
